### ISSUE TITLE
Call getConnectPromise few times

### ISSUE THREAD

**Issue by aborovsky** [gtjoseph/mikronode-ng#4]
Hi, guys!

I have an issue. Trying to reveal: is this a bug, or i'm trying to use current lib in unproper way.
There are some another lib calling time to time  mikronode-ng with different API functions.

**I'm assume, closeOnDone being applied to connection and channel.**

That's why, each time everythins starts with `connection.getConnectPromise()` of the once instantiated connection object.

But it's not working as I assume.
`connection.getConnectPromise()` not being resolved, when called second (and more) time.

There is my test script to reproduce behaviour:

```
var MikroNode = require('../lib/index.js');

var connection = MikroNode.getConnection(process.argv[2], process.argv[3], process.argv[4], {
    closeOnDone: true
});


function getIp() {
    return new Promise(function(resolve) {
        connection.getConnectPromise().then(function(conn) {
            console.log('-------------------- getIp');
            conn.getCommandPromise('/ip/address/print').then(function resolved(values) {
                console.log('Addreses: ' + JSON.stringify(values));
                resolve();
            }, function rejected(reason) {
                console.log('Oops: ' + JSON.stringify(reason));
                resolve();
            });
        });
    });
}


function getArp() {
    return new Promise(function(resolve) {
        connection.getConnectPromise().then(function(conn) {
            console.log('-------------------- getArp');
            conn.getCommandPromise('/ip/address/print').then(function resolved(values) {
                console.log('ARP Addreses: ' + JSON.stringify(values));
                resolve();
            }, function rejected(reason) {
                console.log('Oops: ' + JSON.stringify(reason));
                resolve();
            });
        });
    });
}


getArp().then(getIp);
```

And this is the log output:

```
  mikronode-ng:connection Opening channel: login +0ms
  mikronode-ng:channel Opening channel: login +3ms
  mikronode-ng:channel Channel login idle: writing +3ms
  mikronode-ng:channel Channel login done +39ms
  mikronode-ng:channel Channel login running: pushing command +2ms
  mikronode-ng:channel Channel login more commands +0ms
  mikronode-ng:channel Channel login idle: writing +1ms
  mikronode-ng:channel Channel login done +8ms
  mikronode-ng:channel Channel login closing.  Forced: true +1ms
  mikronode-ng:channel Channel login sending cancel +1ms
  mikronode-ng:channel Channel login closed +0ms
  mikronode-ng:connection Closing channel: login  Channels left: 0 +0ms
  mikronode-ng:connection Resolved +2ms
  mikronode-ng:channel Channel login closing deferred +0ms
-------------------- getArp
  mikronode-ng:connection getCommandPromise +3ms
  mikronode-ng:connection Opening channel: 1 +1ms
  mikronode-ng:channel Opening channel: 1 +0ms
  mikronode-ng:channel Channel 1 idle: writing +1ms
  mikronode-ng:channel Channel 1 done +20ms
  mikronode-ng:channel Channel 1 closing.  Forced: false +2ms
  mikronode-ng:channel Channel 1 closed +0ms
  mikronode-ng:connection Closing channel: 1  Channels left: 0 +1ms
  mikronode-ng:connection Last channel closed +0ms
  mikronode-ng:connection Connection disconnecting: 192.168.2.1 +1ms
ARP Addreses: [...]
  mikronode-ng:connection Connection disconnected: 192.168.2.1 +2ms
```

**Comment by gtjoseph** [comment]
Hi Alexander,

The issue is that whichever function gets called second is trying to create another connect promise on the same connection object.  Since that object is already connected, nothing was done, including the resolve.

I just uploaded and published version 1.0.6 which checks if the connection is already connected and just calls resolve if it is.  I tested it with your scenario and it now works.

Let me know if there are still issues.
